pkg/resources/password_policy.go

```diff
@@ -0,0 +1,447 @@
+	},
+	"min_special_chars": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      1,
+		Description:  "Specifies the minimum number of special characters the password must contain. Supported range: 0 to 256, inclusive. Default: 1",
+		ValidateFunc: validation.IntBetween(0, 256),
+	},
+	"max_age_days": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      90,
+		Description:  "Specifies the maximum number of days before the password must be changed. Supported range: 0 to 999, inclusive. A value of zero (i.e. 0) indicates that the password does not need to be changed. Snowflake does not recommend choosing this value for a default account-level password policy or for any user-level policy. Instead, choose a value that meets your internal security guidelines. Default: 90, which means the password must be changed every 90 days.",
+		ValidateFunc: validation.IntBetween(0, 999),
+	},
+	"max_retries": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      5,
+		Description:  "Specifies the maximum number of attempts to enter a password before being locked out. Supported range: 1 to 10, inclusive. Default: 5",
+		ValidateFunc: validation.IntBetween(1, 10),
+	},
+	"lockout_time_mins": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      15,
+		Description:  "Specifies the number of minutes the user account will be locked after exhausting the designated number of password retries (i.e. PASSWORD_MAX_RETRIES). Supported range: 1 to 999, inclusive. Default: 15",
+		ValidateFunc: validation.IntBetween(1, 999),
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Adds a comment or overwrites an existing comment for the password policy.",
+	},
+}
+
+func PasswordPolicy() *schema.Resource {
+	return &schema.Resource{
+		Description: "A password policy specifies the requirements that must be met to create and reset a password to authenticate to Snowflake.",
+		Create:      CreatePasswordPolicy,
+		Read:        ReadPasswordPolicy,
+		Update:      UpdatePasswordPolicy,
+		Delete:      DeletePasswordPolicy,
+
+		Schema: passwordPolicySchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// CreatePasswordPolicy implements schema.CreateFunc.
+func CreatePasswordPolicy(d *schema.ResourceData, meta interface{}) error {
+	manager, err := snowflake.NewPasswordPolicyManager()
+	if err != nil {
+		return fmt.Errorf("couldn't create password policy builder: %w", err)
+	}
+
+	input := &snowflake.PasswordPolicyCreateInput{
+		PasswordPolicy: snowflake.PasswordPolicy{
+			SchemaObjectIdentifier: snowflake.SchemaObjectIdentifier{
+				Database:   d.Get("database").(string),
+				Schema:     d.Get("schema").(string),
+				ObjectName: d.Get("name").(string),
+			},
+
+			MinLength:           d.Get("min_length").(int),
+			MinLengthOk:         manager.Ok(d.GetOk("min_length")),
```

Comment:
manager.Ok should probably be a helper method in the terraform provider, not SDK